src/vdom.rs

```diff
@@ -258,25 +440,191 @@ where
         }
     }
 
-    fn create(&self, node: Node) {
+    fn create<'a>(&self, registry: &mut EventsRegistry, node: Node<'a>) {
         match node {
             Node::Text(TextNode { text }) => {
                 self.change_list.emit_create_text_node(text);
             }
             Node::Element(ElementNode {
                 tag_name,
+                listeners,
                 attributes,
                 children,
             }) => {
                 self.change_list.emit_create_element(tag_name);
+                for l in listeners {
+                    unsafe {
+                        registry.add(l);
+                    }
+                    self.change_list.emit_new_event_listener(l);
+                }
                 for attr in attributes {
                     self.change_list.emit_set_attribute(&attr.name, &attr.value);
                 }
                 for child in children {
-                    self.create(child.clone());
+                    self.create(registry, child.clone());
                     self.change_list.emit_append_child();
                 }
             }
         }
     }
 }
+
+fn request_animation_frame(f: &Closure<FnMut()>) {
+    web_sys::window()
+        .expect_throw("should have a window")
+        .request_animation_frame(f.as_ref().unchecked_ref())
+        .expect_throw("should register `requestAnimationFrame` OK");
+}
+
+fn with_animation_frame<F>(mut f: F)
+where
+    F: 'static + FnMut(),
+{
+    let g = Rc::new(RefCell::new(None));
+    let h = g.clone();
+
+    let f = Closure::wrap(Box::new(move || {
+        *g.borrow_mut() = None;
+        f();
+    }) as Box<FnMut()>);
+    request_animation_frame(&f);
+
+    *h.borrow_mut() = Some(f);
+}
+
+trait CellOptionExt<T> {
+    fn take(&self) -> Option<T>;
+}
+
+impl<T> CellOptionExt<T> for Cell<Option<T>> {
+    fn take(&self) -> Option<T> {
```

Comment:
I think `cell.take()` is the same as `cell.replace(None)`, right?

Reply:
...yes

good catch

Reply:
And somehow it compiles just fine without this extension trait too -- so there is a `take` method for `Cell<Option<T>>` somewhere already??

Reply:
https://doc.rust-lang.org/nightly/std/cell/struct.Cell.html#method.take

Reply:
how the heck did I miss this when reading over these methods??

Reply:
bahahaha